crates/common/log_manager/src/log_utils.rs

```diff
@@ -1,8 +1,13 @@
+use anyhow::anyhow;
```

Comment:
A library returning `anyhow` errors is highly discouraged. You should be defining concrete error types with the use of `thiserror` instead. Anyhow is usually used at the application level where the errors just need to be reported.

Reply:
IMO there are valid use cases for using `anyhow` in a library crate. What matters the most is the intent, what do we expect users to do. We do not expect code which calls the uploader to behave differently between error variants of e.g. "this does not name a valid file" or "path is not valid unicode", we simply return these errors and show them to the user. We don't have to maintain any Rust API compatibility, we can add or remove error variants at any time.
So I think in this case using an error variant for "file error" and then using `anyhow` for telling the user what exactly went wrong is acceptable.
`anyhow::Error` is also like `Box<dyn Error> + 'static`, but also guarantees `Send` and `Sync`. I originally added it to `InvalidFileName` error variant because i wanted to use either `io::Error` or `Utf8Error` as a source, but using `anyhow!("error details")` directly... I wouldn't say it's an antipattern.

I was recently reading _Zero To Production In Rust_ and there's a free chapter about error handling. There is a dedicated [section about anyhow vs thiserror](https://www.lpalmieri.com/posts/error-handling-rust/#anyhow-or-thiserror), but I'd recommend reading the entire chapter for context.

Reply:
Thanks for the link. Gave me a new perspective. I guess the following is the key point:

```
Library authors cannot (or do not want to) make assumptions on the intent of their users. They steer away from being opinionated (to an extent) - enums give users more control, if they need it.
```

and I completely agree to your following point:

> We do not expect code which calls the uploader to behave differently between error variants of e.g. "this does not name a valid file" or "path is not valid unicode", we simply return these errors and show them to the user.

Here is where I'm wondering what value `anyhow` adds if all we wanted to do was to report the reason to the end-user. Why couldn't it be just a `String` value? Using `anyhow` instead, is just an overkill, right? As the user is less likely to introspect the `source` chain any further, as that was our fundamental assumption when we picked `anyhow` for "opaque" errors in the first place.